gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/project/GradleProject.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.testing.project;
+
+import com.palantir.gradle.testing.files.BuildDir;
+import com.palantir.gradle.testing.files.FileFactory;
+import com.palantir.gradle.testing.files.GradleSourceSet;
+import com.palantir.gradle.testing.files.gradle.GradleFile;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public interface GradleProject extends FileFactory {
+    @Override
+    Path path();
+
+    RootProject rootProject();
+
+    default SubProject subproject(String name) {
+        Path subprojectDir = path().resolve(name);
```

Comment:
something I ran into in GCV, we should enforce that the name provided here is not a project path i.e. `foo:a:b`. Either that or split and create this subproject hierarchy, but I think the former is an easier place to start.